src/include/platform/internal/GenericPlatformManagerImpl_POSIX.cpp

```diff
@@ -64,8 +64,11 @@ template <class ImplClass>
 CHIP_ERROR GenericPlatformManagerImpl_POSIX<ImplClass>::_InitChipStack()
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
+    pthread_mutexattr_t lock_attr;
 
-    mChipStackLock = PTHREAD_MUTEX_INITIALIZER;
+    pthread_mutexattr_init(&lock_attr);
+    pthread_mutexattr_settype(&lock_attr, PTHREAD_MUTEX_RECURSIVE);
+    pthread_mutex_init(&mChipStackLock, &lock_attr);
```

Comment:
Can we by chance move initialization of the lock into the platform manager constructor?  Placing it here means that `_InitChipStack` must be called once and only once.

We've had several issues and PRs that arise simply because `_InitChipStack` was inadvertently called more than once.  It seems like the sane thing to do would be for `_InitChipStack` to be idempotent, where any subsequent calls after the first are a no-op.  That would requiring locking though.

The pattern would be:

* something constructs PlatformManager; now we have a lock
* first _InitChipStack call acquires lock and inits, sets initialized flag
* subsequent `_InitChipStack` calls acquire lock, check initialization flag, return

Does anybody else have thoughts on this?

Reply:
To frame it another way: reinitializing a pthread mutex has undefined behavior.  But yet this happens in a call to a public method on the PlatformManager.  Given that this is public, it seems like the code can and should protect itself from this.

Reply:
@kghost is working on the ChipStack class #5991 which should resolve this issue.